hosting/src/components/TogglePublishDocument.tsx

```diff
@@ -1,34 +1,167 @@
 import {Button} from "@/components/Button";
+import {DatePicker} from "@/components/Form";
+import {Modal} from "@/components/Modal";
 import {useTanamDocument} from "@/hooks/useTanamDocuments";
+import {formatDate} from "@/utils/date";
 import {useParams} from "next/navigation";
+import {useMemo, useState} from "react";
 import {TanamPublishStatus} from "tanam-shared/definitions/TanamPublishStatus";
 
 export function TogglePublishDocument() {
   const {documentId} = useParams<{documentId: string}>() ?? {};
   const {data: document, changeStatus} = useTanamDocument(documentId);
 
+  const maxDate = new Date();
+  maxDate.setDate(maxDate.getDate() + 30);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [isDropdownPublishOpen, setIsDropdownPublishOpen] = useState(false);
+  const [publishedAt, setPublishedAt] = useState<Date | undefined>();
```

Comment:
Update this to "Date | null" for consistency based on requirement

Reply:
Done ✅